Source/WebCore/page/EventHandler.cpp

```diff
@@ -1017,9 +1017,19 @@ void EventHandler::updateSelectionForMouseDrag(const HitTestResult& hitTestResul
             newSelection.setExtent(targetPosition);
     }
 
-    if (m_frame.selection().granularity() != TextGranularity::CharacterGranularity)
+    if (m_frame.selection().granularity() != TextGranularity::CharacterGranularity) {
         newSelection.expandUsingGranularity(m_frame.selection().granularity());
 
+        auto dragStartPosition = m_mousePressNode->renderer()->positionForPoint(m_dragStartPosition, nullptr);
+        VisibleSelection dragStartSelection = dragStartPosition;
+        dragStartSelection.expandUsingGranularity(m_frame.selection().granularity());
+
+        if (!newSelection.isBaseFirst() && !oldSelection.isBaseFirst() && oldSelection.end() < newSelection.end())
+            newSelection.setBase(dragStartSelection.end());
+        else if (newSelection.isBaseFirst() && !oldSelection.isBaseFirst() && oldSelection.start() < newSelection.start())
```

Comment:
Hm...I'm a little bit confused by this check. Is this handling the case where the selection is about to flip? (since the new selection's base is first, and the old selection's base is not first). Or was this intended to be `newSelection.isBaseFirst() && oldSelection.isBaseFirst()`？

Reply:
This is reverting the base back to the start when we're extending selection forwards. If we had adjusted the base in the above code, we need to adjust it back to the start here.